File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        
        if(s.length() != t.length()) return false;
        
        HashMap<Character,Integer> map = new HashMap<>();
        
        
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            
            if(map.containsKey(ch)){
                map.put(ch, map.get(ch)+1);
            }
            else{
                map.put(ch,1);
            }
        }
        
        
        for(int i=0;i<t.length(); i++){
            char ch = t.charAt(i);
            
            
            if(map.containsKey(ch)  && map.get(ch) >= 1){
                
                map.put(ch, map.get(ch)-1);
            }
            else{
                return false;
            }
        }
        
        return true;
        
        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
         sort(s.begin(), s.end());
         sort(t.begin(), t.end());
        
        cout  << s << endl;
        cout << t << endl;
        
        int n1 = s.length();
        int n2 = t.length();
        
        
        if(n1 != n2) return false;
        
        int i=0, j=0;
        
        while(i<n1){
            
            if(s[i] != t[j]) return false;
            
            i++;
            j++;
        }
        
        return true;
    }
};